src/compiler/checker.ts

```diff
@@ -27022,10 +27022,10 @@ namespace ts {
                 case AssignmentDeclarationKind.ExportsProperty:
                 case AssignmentDeclarationKind.Prototype:
                 case AssignmentDeclarationKind.PrototypeProperty:
-                    let valueDeclaration = binaryExpression.left.symbol?.valueDeclaration;
-                    // falls through
                 case AssignmentDeclarationKind.ModuleExports:
-                    valueDeclaration ||= binaryExpression.symbol?.valueDeclaration;
+                    const valueDeclaration = kind !== AssignmentDeclarationKind.ModuleExports

```

Comment:
Was this intended to be equivalent?  It doesn't look equivalent (and it isn't obviously related to `let`/`const`).

Reply:
+1 : This is not equivalent but clearly tests are not failing..

Reply:
It looks equivalent to me... Am I missing something?

Reply:
If it's any of the original three, it'll not be equal to the 4th one, so it picks the first branch, otherwise it's ModuleExports, so it picks the second one. I tried to keep them in order, but maybe it'd be clearer to just not and copy the code.

Reply:
Oh, I see. If it was one of the first few cases but turned out to be undefined, it'll pick the second one. That is different!

Reply:
After reading #43755, I'm sort of thinking this was unintentional and it was just logic reuse, e.g. that this new code is the expected logic. But, maybe @andrewbranch remembers.

Reply:
I checked, and in all of our test cases, if `binaryExpression.left.symbol?.valueDeclaration` is undefined, `binaryExpression.symbol?.valueDeclaration` is also undefined, hence why my PR doesn't fail even though the logic is different.

Reply:
I’m sure I wouldn’t have written this as a falls-through and `||=` if I could have combined the cases like you have here, but it’s disappointing the tests don’t capture that. It’s possible I was being overly defensive.

Reply:
I'm OOF but had some time in the airport, changed it so it's semantically equivalent to be safe.

Reply:
I'm a little worried about someone adding a case and missing the exception, but I agree that it looks equivalent now.

Reply:
The alternative is to just use var, if you all find that preferable.